ambry-api/src/main/java/com/github/ambry/account/ContainerBuilder.java

```diff
@@ -24,13 +27,15 @@
  * in two ways: 1) from an existing {@link Container} object; and 2) by supplying required fields of a {@link Container}.
  * This class is not thread safe.
  */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonPOJOBuilder(withPrefix="set")
 public class ContainerBuilder {
   // necessary
   private short id;
```

Comment:
We should take a careful look at how default values of primitives will behave when using the ContainerBuilder to deserialize json. For example, before an error would be thrown if id was not present in the json. Using jackson, it seems like a `Container` would be created with the default short value of 0 as the id.

Reply:
I initialize the container id and container name with a special static value  UNINITIALIZED_CONTAINER_ID and UNINITIALIZED_CONTAINER_NAME. So that if container id or container name is not present in json string, we can throw an exception. I did the same thing for account id and account name, but not other fields.